lxd/main_nsexec.go

```diff
@@ -31,27 +31,32 @@ package main
 #include <string.h>
 #include <unistd.h>
 
-#define CMDLINE_SIZE (8 * PATH_MAX)
+// External functions
+extern void forkfile();
+extern void forkmount();
+extern void forknet();
+extern void forkproxy();
 
-extern void forkfile(char *buf, char *cur, ssize_t size);
-extern void forkmount(char *buf, char *cur, ssize_t size);
-extern void forknet(char *buf, char *cur, ssize_t size);
-extern void forkproxy(char *buf, char *cur, ssize_t size);
+// Command line parsing and tracking
+#define CMDLINE_SIZE (8 * PATH_MAX)
+char cmdline_buf[CMDLINE_SIZE];
```

Comment:
This very much doesn't look thread-safe to me. This is a global buffer, do we fear that running multiple subcommands from multiple Go-routines will write to the same global buffer?

Reply:
The `cmdline_buf` should be local to the `constructor` imho.

Reply:
Who cares about thread safety here, we run in the constructor before the runtime is spawned, so there are no threads.